### PR TITLE
DOC: Follow repository naming convention

### PR DESCRIPTION
**Pull request by thewtex**
`LabelErodeDilate` to `ITKLabelErodeDilate`

### CODE REVIEW

**Comment by thewtex** [comment]
> 
>    In Circle CI config file: .circleci/config.yml#L4; .circleci/config.yml#L9 .circleci/config.yml#L24; .circleci/config.yml#L25; and this buggy line here (probably introduced by me).
    In setup.py: setup.py#L21; setup.py#L23; and setup.py#L53 should may be also changed to the source code GitHub url (i.e. https://github.com/InsightSoftwareConsortium/ITKLabelErodeDilate/)

Good catch - done!

File: .circleci/config.yml [hunk 1]
```diff
@@ -17,12 +17,12 @@ jobs:
             SHASNIP=$(echo $CIRCLE_SHA1 | cut -c1-7)
             cat > dashboard.cmake << EOF
             set(CTEST_SITE "CircleCI")
-            set(CTEST_BUILD_NAME "External-LabelErodeDilate-${CIRCLE_BRANCH}-${CIRCLE_BUILD_NUM}-${SHASNIP}")
+            set(CTEST_BUILD_NAME "External-ITKLabelErodeDilate-${CIRCLE_BRANCH}-${CIRCLE_BUILD_NUM}-${SHASNIP}")
```
Comment (dzenanz) [inline .circleci/config.yml]: Changes in this file are not consistent with the ModuleTemplate:
https://github.com/InsightSoftwareConsortium/ITKModuleTemplate/blob/bb692c7526016ff3729eda2a2f05b29b160f338e/%7B%7Bcookiecutter.project_name%7D%7D/.circleci/config.yml#L20

The name of the repository is `ITKLabelErodeDilate`, as it should be. And http(s) links to it need to reflect that. But the name of the module is `LabelErodeDilate`, as external modules do not have 'ITK` prefix as part of their name.

File: setup.py [hunk 2]
```diff
@@ -18,9 +18,9 @@
     author_email='Richard.Beare@med.monash.edu.au',
     packages=['itk'],
     package_dir={'itk': 'itk'},
-    download_url=r'https://github.com/InsightSoftwareConsortium/LabelErodeDilate',
+    download_url=r'https://github.com/InsightSoftwareConsortium/ITKLabelErodeDilate',
     description=r'An ITK module for erode and dilate operations on label images',
-    long_description='LabelErodeDilate provides classes for morphological math'
+    long_description='ITKLabelErodeDilate provides classes for morphological math'
```
Comment (dzenanz) [inline setup.py]: Shouldn't this be LabelErodeDilate without ITK prefix?

File: .circleci/config.yml (hunk 1 above)
Comment (thewtex) [reply .circleci/config.yml]: > Changes in this file are not consistent with the ModuleTemplate:
https://github.com/InsightSoftwareConsortium/ITKModuleTemplate/blob/bb692c7526016ff3729eda2a2f05b29b160f338e/%7B%7Bcookiecutter.project_name%7D%7D/.circleci/config.yml#L20

This is using `cookiecutter.project_name` (contains *ITK*) versus `cookiecutter.module_name`, which does not.  The defaults are computed here:

https://github.com/InsightSoftwareConsortium/ITKModuleTemplate/blob/bb692c7526016ff3729eda2a2f05b29b160f338e/cookiecutter.json#L5-L6

File: setup.py (hunk 2 above)
Comment (thewtex) [reply setup.py]: The project name is appropriate here, and the *ITK* prefix helps make the connection with ITK.

**Comment by thewtex** [comment]
TravisCI failures are due to `transfer.sh` being down.

**Comment by jhlegarreta** [comment]
Thanks Matt.

**Pull request merged**